server/src/test/java/org/uiautomation/ios/e2e/nativegestures/ScrollTest.java

```diff
@@ -15,44 +15,107 @@
 
 import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
 import org.testng.Assert;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 import org.uiautomation.ios.BaseIOSDriverTest;
 import org.uiautomation.ios.SampleApps;
 import org.uiautomation.ios.UIAModels.ScrollDirection;
+import org.uiautomation.ios.UIAModels.UIAElement;
+import org.uiautomation.ios.UIAModels.UIAStaticText;
+import org.uiautomation.ios.UIAModels.predicate.*;
 import org.uiautomation.ios.client.uiamodels.impl.RemoteIOSDriver;
 import org.uiautomation.ios.client.uiamodels.impl.RemoteUIAScrollView;
 
 
 public class ScrollTest extends BaseIOSDriverTest {
 
+  private static final String verView1 = "Vertical View 1";
+  private static final String verView2 = "Vertical View 2";
+  private static final String verView3 = "Vertical View 3";
+
+  private static final String horView1 = "Horizontal View 1";
+  private static final String horView2 = "Horizontal View 2";
+  private static final String horView3 = "Horizontal View 3";
+
   @BeforeClass
   public void startDriver() {
     driver = new RemoteIOSDriver(getRemoteURL(), SampleApps.ppNQASampleAppCap());
   }
 
+  private void findElement(String name) {
+    Criteria c1 = new TypeCriteria(UIAStaticText.class);
+    Criteria c2 = new NameCriteria(name, MatchingStrategy.starts);
+    Criteria c = new AndCriteria(c1, c2);
+    UIAElement element = driver.findElement(c);
+  }
+
   @Test
   public void testVerticalScrollingDown() {
     Assert.assertTrue(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 1')]").isDisplayed());
-    Assert.assertFalse(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 2')]").isDisplayed());
-    Assert.assertFalse(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 3')]").isDisplayed());
+
+    // no Vertical View 2
+    try {
+      findElement(verView2);
+      Assert.fail("shouldn't find element" + verView2);
+    } catch (NoSuchElementException e) {
+      // expected
+    }
+
+    // no Vertical View 3
+    try {
+      findElement(verView3);
+      Assert.fail("shouldn't find element" + verView3);
+    } catch (NoSuchElementException e) {
+      // expected
+    }
 
     ((RemoteUIAScrollView) driver.findElementByXPath("//UIAScrollView[contains(@name, 'Vertical Name')]")).scroll(ScrollDirection.DOWN);
 
     waitForElement(driver, By.xpath("//UIAStaticText[contains(@name, 'Vertical View 2')]"), 6);
     Assert.assertTrue(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 2')]").isDisplayed());
-    Assert.assertFalse(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 1')]").isDisplayed());
-    Assert.assertFalse(driver.findElementByXPath("//UIAStaticText[contains(@name, 'Vertical View 3')]").isDisplayed());
+
+    // no Vertical View 1
+    try {
+      findElement(verView1);
```

Comment:
Another option you have to get rid of the `try {` condition .. fail `} catch {` pass `}` code is to put the `try` .. `catch` in the `private` method `findElement` and change the return to of `findElement` to `boolean`  -- `return true` in the `catch` . Thus, making all the calls from the test more readable 

```java 
 Assert.assertTrue(findElement(verView2), "Condition should not be true!");
```

BTW, In case you are wondering why I'm asking for a change here -- While the code works as written, it doesn't really pass this "sniff test" -- https://en.wikipedia.org/wiki/Code_smell In TestNG whenever, there is a `try {` condition .. fail `} catch {` pass `}` OR the use of `Assert.fail()` a reviewer will almost always raise a yellow flag here.